src/compiler/transform/Story/props.ts

```diff
@@ -0,0 +1,257 @@
+  );
+}
+
+function insertDescriptionStory({
+  comment,
+  currentDocsProperty,
+  filename,
+}: {
+  comment: Comment;
+  currentDocsProperty: Property;
+  filename?: string;
+}) {
+  if (currentDocsProperty.value.type !== 'ObjectExpression') {
+    // TODO: Update error message
+    throw new Error('Invalid schema');
+  }
+
+  const findDescriptionIndex = () =>
+    findPropertyIndex('description', currentDocsProperty.value as ObjectExpression);
+
+  if (findDescriptionIndex() === -1) {
+    currentDocsProperty.value.properties.push(
+      createProperty('description', createObjectExpression([]))
+    );
+  }
+
+  const currentDescriptionProperty = currentDocsProperty.value.properties[
+    findDescriptionIndex()
+  ] as Property;
+
+  if (currentDescriptionProperty.value.type !== 'ObjectExpression') {
+    // TODO: Update error message
+    throw new Error('Invalid schema');
+  }
+
+  const findStoryIndexIndex = () =>
+    findPropertyIndex('story', currentDescriptionProperty.value as ObjectExpression);
+
+  if (findStoryIndexIndex() !== -1) {
+    logger.warn(
+      `One of <Story /> component(s) already has explictly set description. Ignoring the HTML comment above. Stories file: ${filename}`
+    );
+
+    return;
+  }
+
+  currentDescriptionProperty.value.properties.push(
+    createProperty('story', {
+      type: 'Literal',
+      value: dedent(comment.data),
+    })
+  );
+
+  currentDocsProperty.value.properties[findDescriptionIndex()] = currentDescriptionProperty;
+}
+
+function insertSourceCode({
+  component,
+  currentDocsProperty,
+  filename,
+  originalCode,
+}: {
+  component: Component;
+  currentDocsProperty: Property;
+  filename?: string;
+  originalCode: string;
+}) {
+  if (currentDocsProperty.value.type !== 'ObjectExpression') {
+    // TODO: Update error message
+    throw new Error('Invalid schema');
+  }
+
+  const findSourceIndex = () =>
+    findPropertyIndex('source', currentDocsProperty.value as ObjectExpression);
+
+  if (findSourceIndex() === -1) {
+    currentDocsProperty.value.properties.push(createProperty('source', createObjectExpression([])));
+  }
+
+  const currentSourceProperty = currentDocsProperty.value.properties[findSourceIndex()] as Property;
+
+  if (currentSourceProperty.value.type !== 'ObjectExpression') {
+    // TODO: Update error message
+    throw new Error('Invalid schema');
+  }
+
+  const findStoryIndex = () =>
+    findPropertyIndex('story', currentSourceProperty.value as ObjectExpression);
+
+  if (findStoryIndex() !== -1) {
+    // TODO: That's not unepected, curious or anything. No need for a warning, right?
+    return;
+  }
+
+  const { fragment, start, end } = component;
+  // TODO: From there we need to start the logic for filling Story's children raw source.
+  // Temporarily used the whole raw soure fragment.
+  const { nodes } = fragment;
+  const codeValue = originalCode.slice(start, end);
```

Comment:
This module is lenghty. Need to split into two, because there's going to be added more logic related to the `source.

Comment for reference: https://github.com/storybookjs/addon-svelte-csf/pull/181#issuecomment-2143539873

From this point, what I need is to start the extraction logic _(step 2)_ of the `<Story />`. And determine it's children's content - including empty ones and from `setTemplate` - whose AST node should exist the `nodes.compiled.setTemplateSnippet`.

And then, it should be an easier path to continue with step 4 at runtime.